src/systems/systems/src/enhanced_gpwc/navigation_display.rs

```diff
@@ -0,0 +1,65 @@
+use crate::simulation::{
+    InitContext, Read, SimulationElement, SimulatorReader, SimulatorWriter, VariableIdentifier,
+    Write,
+};
+use uom::si::{
+    f64::{Length, Ratio},
+    length::nautical_mile,
+    ratio::percent,
+};
+
+pub struct NavigationDisplay {
+    range_knob_id: VariableIdentifier,
+    range_knob_position: usize,
+    range: Length,
+    mode_id: VariableIdentifier,
+    mode: u8,
+    terrain_on_nd_pb_id: VariableIdentifier,
+    terrain_on_nd_pb_active: bool,
+    terrain_on_nd_active: bool,
+    potentiometer_id: VariableIdentifier,
+    potentiometer: Ratio,
+    // output variables of the display
+    egpwc_nd_range_id: VariableIdentifier,
+    egpwc_nd_terrain_active_id: VariableIdentifier,
```

Comment:
see rust guidelines.md and readme.md ideally systems should not be aware they are running in a simulator and simvar state should be passed and vars not being read here i.e. the system is unaware that it is running in a simulator

Reply:
Where should I read these variables?

Reply:
Resolved in development channel on discord